### DISCUSSION TITLE
How to use handler.reject in onError of QueuedInterceptor

### DISCUSSION THREAD

**Discussion by jackie-weiwei** [cfug/dio#2093] [General]
### Request Statement

flutter version: 3.10.4
dio version : 5.4.0

When I call handler.reject(DioException(requestOptions: options)) in onError, The caller displays the following error：
```
Exception has occurred.
DioException (DioException [unknown]: null)
```
![image](https://github.com/cfug/dio/assets/101157197/6bd8ac2c-fdd3-449a-abd8-76d4b0ef8c35)

I don't understand now why the caller reports an error after rejecting. How should I use reject? Please help. Thank you.
my code like this: 
```dart
Response res = await RestApi.post(
    path: "/test/test", data: {"mobile": "1000000000"});
if (res == null) {
  print("请求失败");
}
```

interceptor code like this:
```dart
    _dio.interceptors.add(QueuedInterceptorsWrapper(
      onRequest: (options, handler) async {
        options.headers["access"] = _token;
        return handler.next(options); //continue
      },
      onResponse: (response, handler) {
        String path = response.requestOptions.path;
        if (path.contains("login")) {
          _token = response.data["accessToken"];
          _token2 = response.data["refreshToken"];
        }
        debugPrint(path);
        return handler.next(response); // continue
      },
      onError: (error, handler) async {
        if (error.response?.statusCode == 401) {
          final options = error.response!.requestOptions;
          bool bSucc = await refreshToken();

          if (bSucc) {
            var res = await _dio.fetch(error.response!.requestOptions);
            handler.resolve(res); //continue
          } else {
            handler.reject(DioException(requestOptions: options));
          }
        } else {
          handler.next(error);
        }
      },
    ));
```

**Comment by jackie-weiwei** [comment]
@AlexV525 can you help me please

**Comment by AlexV525** [comment]
https://pub.dev/documentation/dio/latest/dio/ResponseInterceptorHandler/reject.html

As the doc mentioned, the method will reject the response with an error.

**Reply by jackie-weiwei** [reply]
I used it according to the documentation, but the caller still reported an error. Maybe I still don't quite understand what the problem is?
@AlexV525

**Reply by AlexV525** [reply]
`reject` should throw the exception you've rejected, it is expected.

**Reply by jackie-weiwei** [reply]
How to intercept DioExceptionType.connectionError. I tried to intercept it in QueuedInterceptor's Onerror, but it was not successful. My doubt is that onerror intercepts error messages, so all error messages should trigger this event, but this is not the case. May I ask why?
